sdk/signalr/Microsoft.Azure.WebJobs.Extensions.SignalRService/src/Config/SignalROptions.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using Microsoft.Azure.SignalR;
+using Microsoft.Azure.SignalR.Management;
+using Microsoft.Extensions.Configuration;
+
+namespace Microsoft.Azure.WebJobs.Extensions.SignalRService
+{
+    /// <summary>
+    /// Configuration options for SignalR extensions.
+    /// </summary>
+    /// <remarks>
+    /// Most properties here can be configured in <see cref="IConfiguration"/>, except JSON serialization.
+    /// </remarks>
+    public class SignalROptions
+    {
+        /// <summary>
+        /// Gets or sets the service endpoints.
+        /// </summary>
+        public ServiceEndpoint[] ServiceEndpoints { get; set; }
```

Comment:
Typically we don't have settable collections, instead we have a list with a getter that can be mutated, e.g. https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/servicebus/Azure.Messaging.ServiceBus/src/Processor/ServiceBusSessionProcessorOptions.cs#L151